nixos/modules/security/pam.nix

```diff
@@ -218,7 +218,7 @@ let
       # Samba stuff to the Samba module.  This requires that the PAM
       # module provides the right hooks.
```

Comment:
Thanks @lethalman, an assertion would be a good addition to a get clearer error messages. But I would still enforce unixAuth = true if enableEcryptfs or pamMount is set.